src/hybrid.jl

```diff
@@ -3,67 +3,77 @@
 
 Train NB model with discrete and continuous features by estimating P(x⃗|c)
 """
-function fit{C, T<: AbstractFloat, U <: Integer}(model::HybridNB, continuous_features::Vector{Vector{T}}, discrete_features::Vector{Vector{U}}, labels::Vector{C})
+function fit{C, T<: AbstractFloat, U <: Integer, N}(model::HybridNB, continuous_features::Dict{N, Vector{T}}, discrete_features::Dict{N, Vector{U}}, labels::Vector{C})
+    A = 1.0/float(length(labels))
     for class in model.classes
         inds = find(labels .== class)
-        for (j, feature) in enumerate(continuous_features)
+        model.priors[class] = A*float(length(inds))
+        for (name, feature) in continuous_features
             f_data = feature[inds]
-            model.c_kdes[class][j] = InterpKDE(kde(f_data[isfinite(f_data)]), eps(Float64), InterpLinear)
+            model.c_kdes[class][name] = InterpKDE(kde(f_data[isfinite(f_data)]), eps(Float64), InterpLinear)
         end
-        for (j, feature) in enumerate(discrete_features)
+        for (name, feature) in discrete_features
             f_data = feature[inds]
-            model.c_discrete[class][j] = ePDF(f_data[isfinite(f_data)])
+            model.c_discrete[class][name] = ePDF(f_data[isfinite(f_data)])
         end
     end
     return model
 end
 
+"""
+    train(HybridNB, continuous, discrete, labels) -> model2
+"""
+function train{C, T<: AbstractFloat, U <: Integer, N}(::Type{HybridNB}, continuous_features::Dict{N, Vector{T}}, discrete_features::Dict{N, Vector{U}}, labels::Vector{C})
+    return fit(HybridNB(labels, N), continuous_features, discrete_features, labels)
+end
+
 
 """
     fit(m::HybridNB, f_c::Matrix{Float64}, labels::Vector{Int64})
 
 Train NB model with continuous features only
 """
 function fit{C, T<: AbstractFloat}(model::HybridNB, continuous_features::Matrix{T}, labels::Vector{C})
-    discrete_features = Vector{Vector{Int64}}()
+    discrete_features = Dict{Symbol, Vector{Int64}}()
     return fit(model, restructure_matrix(continuous_features), discrete_features, labels)
 end
 
 
 """computes log[P(x⃗ⁿ|c)] ≈ ∑ᵢ log[p(xⁿᵢ|c)] """
-function sum_log_x_given_c!{T <: AbstractFloat, U <: Integer}(class_prob::Vector{Float64}, feature_prob::Vector{Float64}, m::HybridNB, continuous_features::Vector{Vector{T}}, discrete_features::Vector{Vector{U}}, c)
+function sum_log_x_given_c!{T <: AbstractFloat, U <: Integer, N}(class_prob::Vector{Float64}, feature_prob::Vector{Float64}, m::HybridNB, continuous_features::Dict{N, Vector{T}}, discrete_features::Dict{N, Vector{U}}, c)
     for i = 1:num_samples(m, continuous_features, discrete_features)
-
-        for j = 1:num_kdes(m)
-            if isnan(continuous_features[j][i])
+        for (j, name) in enumerate(keys(continuous_features))
+            x_i = continuous_features[name][i]
+            if isnan(x_i)
```

Comment:
maybe replace this `if` statement with:

`feature_prob[j] = isnan(x_i) ? NaN : pdf(m.c_kdes[c][name], x_i)`